mod/blog/elgg-plugin.php

```diff
@@ -14,6 +14,57 @@
 		'blog/auto_save_revision' => [],
 		'blog/delete' => [],
 	],
+	'routes' => [
+		'blog_owner' => [
+			'path' => '/blog/owner/{username?}',
+			'resource' => 'blog/owner',
+		],
+		'blog_friends' => [
+			'path' => '/blog/friends/{username?}',
+			'resource' => 'blog/friends',
+		],
+		'blog_archive' => [
+			'path' => '/blog/archive/{username?}/{lower?}/{upper?}',
+			'resource' => 'blog/archive',
+			'requirements' => [
+				'lower' => '\d+',
+				'upper' => '\d+',
+			],
+		],
+		'blog_view' => [
```

Comment:
For `view` we probably need some system-wide approach, so we can resolve URLs to entities.